crates/ruff_linter/src/rules/refurb/rules/verbose_decimal_constructor.rs

```diff
@@ -152,14 +152,29 @@ pub(crate) fn verbose_decimal_constructor(checker: &mut Checker, call: &ast::Exp
             let Some(float) = float.as_string_literal_expr() else {
                 return;
             };
+
+            let normalized_float_string = float.value.to_str().trim().to_lowercase();
+
             if !matches!(
-                float.value.to_str().to_lowercase().as_str(),
-                "inf" | "-inf" | "infinity" | "-infinity" | "nan"
+                normalized_float_string.as_str(),
+                "inf"
+                    | "+inf"
+                    | "-inf"
+                    | "infinity"
+                    | "+infinity"
+                    | "-infinity"
+                    | "nan"
+                    | "+nan"
+                    | "-nan"
             ) {
                 return;
             }
```

Comment:
Unrelated to your fix: To avoid an allocation in many cases, it might be worth trying an exact match first before calling `to_lowercase`.

```rust
let trimmed = float.value.to_str().trim();

if is_inf_infinity_or_nan(trimmed) {
	return;
}

let normalized_float_string = trimmed.to_lowercase();

if is_inf_infinity_or_nan(&normalized_flaot_string) {
	return;
}
```

Reply:
Maybe I'm misunderstanding but I think we have to allocate anyway: we only get to escape early here if `float.value` is _not_ inf, infinity, or nan (after trimming and lowercasing). So if we first checked that the trimmed version wasn't "nan" etc., we would still have to rule out the case where it's "NaN".

Reply:
Yes, it's just that we do it less often. I guess the overall idea is to spell out all common variations so that the rule can short-circuit for almost all valid-cases and only allocates for uncommon spellings (e.g. `nAn`).

Reply:
Ok, I think the latest commit avoids an allocation for this check in all cases - let me know if you think it's a bit too awkward or if there's a better way 😄 I wanted to avoid using a regex or any other heavy machinery since it seemed a bit overpowered for this case.

Reply:
(Gonna merge this in to ever-so-slightly ease your code review burden... if you wake up in a cold sweat I'll make a followup PR 😄 )